### PR TITLE
mac autoconf: support current and future SDKs

### LINKED ISSUE #378: Change MacOS build mmacosx-version-min to 10.5
PR #356 removes deprecated support for 10.4 and adds support for the newer CoreAudio APIs.
After it merges, we should update configure.in and change

    -mmacosx-version-min=10.4
to

    -mmacosx-version-min=10.5

### LINKED ISSUE #468: Mac: runtime crashes on Big Sur if built using SDK 11
Here is an example error from #395
Error: ||PaMacCore (AUHAL)|| Error on line 1277: err='-66748', msg=Unknown Error

This crash was conflated with Issue #218, which involved deprecated audio APIs. 
But this crash is happening even after the deprecated audio calls have been replaced in PR #356.

jmelas wrote in [#218 comment](https://github.com/PortAudio/portaudio/issues/218#issuecomment-731310548)
> portaudio fails in runtime on macOS 11 only if the program is built against macOS SDK 11.
> Latest XCode 12.2 comes with macOS SDK 11.
> I downgraded Xcode to version 12.1 which uses 10.15 SDK and then portaudio worked fine on Big Sur!

Related Issues:
#395 - Can't open stream on macOS 11.x 
https://github.com/gordonklaus/portaudio/issues/41
https://github.com/tlecomte/friture/issues/154

### PR DESCRIPTION
**Pull request by philburk**
Improve the way we generate the path to the SDK so that it
it not tied to specific versions.

Also changed macosx-version-min to 10.6

Fixes #378
Fixes #468

### CODE REVIEW

**Comment by Be-ing** [comment]
I assume you've tested this locally on macOS 11? It doesn't seem that you're running any macOS CI??

**Comment by philburk** [comment]
Yes, I tested locally. We should add some Mac CI.

**Comment by Be-ing** [comment]
I already added macOS CI in #459.

**Comment by Be-ing** [comment]
This change looks good to me. Is there anything else that needs to be taken care of before releasing?

File: configure [hunk 1]
```diff
@@ -15852,40 +15840,11 @@ case "${host_os}" in
 
            *)
                                                                                     if xcodebuild -version -sdk macosx10.5 Path >/dev/null 2>&1 ; then
-                 mac_version_min="-mmacosx-version-min=10.3"
+                 mac_version_min="-mmacosx-version-min=10.5"
```
Comment (Be-ing) [inline configure:15843]: Did you mean 10.6?

File: configure.in [hunk 2]
```diff
@@ -235,40 +235,11 @@ case "${host_os}" in
               dnl what SDKs are available and to get the full path of
               dnl the SDKs.
               if xcodebuild -version -sdk macosx10.5 Path >/dev/null 2>&1 ; then
-                 mac_version_min="-mmacosx-version-min=10.3"
+                 mac_version_min="-mmacosx-version-min=10.5"
```
Comment (Be-ing) [inline configure.in:238]: Did you mean 10.6?

**Comment by philburk** [comment]
@jmelas - I see that in #359 you suggested:

    mac_sysroot="-isysroot $(xcrun --sdk macosx --show-sdk-path)"

How does that compare to using xcodebuild like I did in this script?

File: configure (hunk 1 above)
Comment (philburk) [reply configure:15843]: Not if the SDK is 10.5

Comment (philburk) [reply configure:15843]: This file is auto generated.

Comment (Be-ing) [reply configure:15843]: ah, thanks for clarifying. I generally recommend against putting autogenerated files into version control

File: configure.in (hunk 2 above)
Comment (philburk) [reply configure.in:238]: The SDK is 10.5 so the min is 10.5

**Comment by jmelas** [comment]
> @jmelas - I see that in #359 you suggested:
> 
> ```
> mac_sysroot="-isysroot $(xcrun --sdk macosx --show-sdk-path)"
> ```
> 
> How does that compare to using xcodebuild like I did in this script?

Hi @philburk 
In my machine the command 
`xcrun --sdk macosx --show-sdk-path`
executes much faster than
`xcodebuild -version -sdk macosx Path`

**Comment by philburk** [comment]
Thanks jmelas. I switched to xcrun.

File: configure [hunk 3]
```diff
@@ -723,7 +723,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
```
Comment (RossBencina) [inline configure:726]: It is unclear from the commit message why any changes are being made to this `runstatedir`. Are these changes (and related ones below) needed for the mac build to work?

If these are unrelated changes I request that they be moved to a separate PR.

**Review by RossBencina — APPROVED** [review]
I can't easily test these changes right now, but if you have multiple people saying that the changes work, then I see no reason to hold off on merging.

File: configure (hunk 3 above)
Comment (philburk) [reply configure:726]: I did not not notice these. It was not intentional.
Maybe this was a difference from running autoconf on Linux vs Mac. I will check.

Comment (philburk) [reply configure]: The Linux autoconf generates those lines. The Mac does not.
Both configure scripts seems to run fine on both platforms.
But to minimize unnecessary changes, I will merge the Linux generated configure.
